### PR TITLE
TINKERPOP-2063 Add `Tree` support to the GLVs (Python, .NET, Go, JavaScript)

### PR DESCRIPTION
**Pull request by GumpacG**
## Summary

This PR brings a matching native `Tree` type to all Gremlin Language Variants so that a server-produced
`tree()` result deserializes into a first-class object with the same read/navigation
API instead of an untyped map (or, in JavaScript, a serializer stub that threw).

The work is deserialization + read-navigation focused: trees are computed on the
server and streamed to the client, so each GLV gains (1) a native `Tree` type and
(2) a GraphBinary `0x2b` (de)serializer. A symmetric serialize path is included for
round-trip/test symmetry.

## What changed per variant

- gremlin-python: `Tree` class in `structure/graph.py`; `TreeIO` in `graphbinaryV4.py`
  (auto-registered for `0x2b`); removed the `@StepTree` feature-test skip.
- gremlin-dotnet: `Structure/Tree.cs` and `GraphBinary4/Types/TreeSerializer.cs`;
  registered `DataType.Tree`; implemented the Gherkin tree-structure assertion; removed
  the `@StepTree` skip.
- gremlin-go: `driver/tree.go` plus `treeType` (`0x2b`), `readTree`/`treeWriter`, and
  registration; removed the `~@StepTree` cucumber tag.
- gremlin-javascript: `Tree` class in `structure/graph.ts`; `TreeSerializer.js`
  replacing the previous stub; exported `Tree`; removed the named tree ignores.

## Testing

- Per-GLV unit tests covering the full read API, structural equality, GraphBinary
  round-trips (nested, empty, null-key), duplicate-sibling-key collapse, element-key
  equality (by id and concrete type), `childAt` on a missing key, `addTree` null
  rejection, `splitParents` edge cases, and `getLeafTrees`.
- Per-GLV integration tests run `g.V(1).out().out().tree()` against the modern graph
  and navigate the native result.
- The shared Gherkin `tree()` feature scenarios were un-skipped in each GLV and assert
  tree structure through the read API.

## Notes

- Scope is the GLVs; the Java core change shipped in #3448.
- `Tree` results travel server -> client; the serialize path exists for symmetry and
  round-trip tests rather than a common workflow.

https://issues.apache.org/jira/browse/TINKERPOP-2063

Assisted-by: Kiro: Claude Opus 4.8

### CODE REVIEW

File: gremlin-dotnet/test/Gremlin.Net.IntegrationTest/Gherkin/GherkinTestRunner.cs [hunk 1]
```diff
@@ -113,12 +113,6 @@ public void RunGherkinBasedTests(IMessageSerializer messageSerializer)
                         continue;
                     }
 
-                    if (scenario.Tags.Concat(feature.Tags).Any(t => t.Name == "@StepTree"))
-                    {
-                        failedSteps.Add(scenario.Steps.First(), new IgnoreException(IgnoreReason.TreeStepNotSupported));
```
Comment (Cole-Greer) [inline gremlin-dotnet/test/Gremlin.Net.IntegrationTest/Gherkin/GherkinTestRunner.cs:122]: Nit: Now that we aren't using it here, I think we should entirely remove [IgnoreReason.TreeStepNotSupported](https://github.com/apache/tinkerpop/blob/5aae2577c29c7af7e295ef0860eb6eb5d4816cb8/gremlin-dotnet/test/Gremlin.Net.IntegrationTest/Gherkin/IgnoreException.cs#L65-L68). It's a minor cleanup and will verify no one else is misusing it.

**Comment by Cole-Greer** [comment]
VOTE +1 (pending the minor cleanup above)

**Comment by kenhuuu** [comment]
VOTE +1

File: gremlin-dotnet/test/Gremlin.Net.IntegrationTest/Gherkin/GherkinTestRunner.cs (hunk 1 above)
Comment (GumpacG) [reply gremlin-dotnet/test/Gremlin.Net.IntegrationTest/Gherkin/GherkinTestRunner.cs:122]: Good catch. Removed.
